.github/workflows/publish-npm.yml

```diff
@@ -0,0 +1,62 @@
+name: Publish to NPM
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "21"
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Install Rust
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+          source $HOME/.cargo/env
+
+      - name: Install Solana CLI
+        run: |
+          sh -c "$(curl -sSfL https://release.solana.com/v1.18.15/install)"
+          export PATH="/home/runner/.local/share/solana/install/active_release/bin:$PATH"
+          solana --version
+
+      - name: Install Anchor CLI
+        run: |
+          cargo install --git https://github.com/coral-xyz/anchor --tag v0.30.0 anchor-cli --locked
+          anchor --version
```

Comment:
nit: There's a [github action](https://github.com/metaDAOproject/setup-anchor) that handles all this:
```
steps:
  - uses: actions/checkout@v3
  - uses: metadaoproject/setup-anchor@v2
    with: 
      anchor-version: '0.28.0' 
      solana-cli-version: '1.14.20'
      node-version: '16.15.1'
```

Reply:
8f0c445